lib/AST/Decl.cpp

```diff
@@ -5703,12 +5704,31 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
     if (FD && !FD->isMutating() && !FD->isImplicit() && FD->isInstanceMember()&&
         !FD->getDeclContext()->getDeclaredInterfaceType()
                  ->hasReferenceSemantics()) {
-      // Do not suggest the fix it in implicit getters
+      // Do not suggest the fix-it in implicit getters
       if (auto AD = dyn_cast<AccessorDecl>(FD)) {
         if (AD->isGetter() && !AD->getAccessorKeywordLoc().isValid())
           return;
+
+        // Do not suggest the fix-it if we have an implicitly nonmutating
+        // setter in a protocol extension and we're assigning to a mutating
+        // protocol requirement.
```

Comment:
This is very specific but still feels like it's not getting at the interesting bit. How about just "if the `Self` type (`AD->getDeclContext()->getSelfInterfaceType()`) is class-bound (`isAnyClassReferenceType()`)?"

Reply:
Sure, do you mean we should replace all of the checks below with a simple "is Self type class-bound or not"?

Reply:
Yeah. If Self is class-bound, you can't insert `mutating`, at least not today, and that covers all the cases we care about introduced by your other change.

Reply:
Alright, done!